lib/trino-filesystem-cache-alluxio/src/main/java/io/trino/filesystem/alluxio/TracingCacheManager.java

```diff
@@ -0,0 +1,162 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.alluxio;
+
+import alluxio.client.file.CacheContext;
+import alluxio.client.file.cache.CacheManager;
+import alluxio.client.file.cache.CacheUsage;
+import alluxio.client.file.cache.PageId;
+import alluxio.exception.PageNotFoundException;
+import alluxio.file.ReadTargetBuffer;
+import alluxio.network.protocol.databuffer.DataFileChannel;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+
+import java.nio.ByteBuffer;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static io.trino.filesystem.alluxio.AlluxioTracing.withTracing;
+import static io.trino.filesystem.cache.CacheSystemAttributes.CACHE_FILE_READ_POSITION;
+import static io.trino.filesystem.cache.CacheSystemAttributes.CACHE_FILE_READ_SIZE;
+import static io.trino.filesystem.cache.CacheSystemAttributes.CACHE_FILE_WRITE_POSITION;
+import static io.trino.filesystem.cache.CacheSystemAttributes.CACHE_FILE_WRITE_SIZE;
+import static io.trino.filesystem.cache.CacheSystemAttributes.CACHE_KEY;
+import static java.util.Objects.requireNonNull;
+
+public class TracingCacheManager
+        implements CacheManager
+{
+    private final Tracer tracer;
+    private final String cacheKey;
+    private final CacheManager delegate;
+
+    public TracingCacheManager(Tracer tracer, String cacheKey, CacheManager delegate)
+    {
+        this.tracer = requireNonNull(tracer, "tracer is null");
+        this.cacheKey = requireNonNull(cacheKey, "cacheKey is null");
+        this.delegate = requireNonNull(delegate, "delegate is null");
+    }
+
+    @Override
+    public boolean put(PageId pageId, ByteBuffer byteBuffer, CacheContext cacheContext)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.put")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .setAttribute(CACHE_FILE_WRITE_POSITION, (long) byteBuffer.position())
+                .setAttribute(CACHE_FILE_WRITE_SIZE, (long) byteBuffer.remaining())
+                .startSpan();
+        return withTracing(span, () -> delegate.put(pageId, byteBuffer, cacheContext));
+    }
+
+    @Override
+    public int get(PageId pageId, int position, int length, ReadTargetBuffer readTargetBuffer, CacheContext cacheContext)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.get")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .setAttribute(CACHE_FILE_READ_POSITION, pageId.getPageIndex() * length)
```

Comment:
Should be position.

Reply:
position will be zero for a given page.

Reply:
At https://github.com/trinodb/trino/blob/8807ef2f1d245c045e6254559dff52aea11881bc/lib/trino-filesystem-cache-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioInputHelper.java#L128 the `currentPageOffset` might not be zero?

Reply:
I should have used page size to calculate position properly. Fixing